main.nf

```diff
@@ -140,392 +165,666 @@ process fastqc {
-    write.table(hmap\$carpet, 'log2CPM_sample_distances.txt', quote=FALSE, sep="\t")
-    
+    write.table(hmap\$carpet, 'log2CPM_sample_distances.txt', quote=FALSE, sep="\\t")
+
     file.create("corr.done")
+    #Printing sessioninfo to standard out
+    print("Sample correlation info:")
+    sessionInfo()
     """
 }
 
 
-*/
+/*
+ * STEP 5.1 - Bowtie 2 against reference genome
+ */
+process bowtie2 {
+
+    tag "$name"
+
+    module 'bioinfo-tools'
+    module 'bowtie2'
+    module 'samtools'
+
+    cpus 4
+    memory { 32.GB * task.attempt }
+    time { 120.h * task.attempt }
+    errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+    maxRetries 3
+    maxErrors '-1'
+
+    publishDir "${params.outdir}/bowtie2", mode: 'copy'
+
+    input:
+    file(reads:'*') from trimmed_reads_bowtie2
+    set val(name) from name_for_bowtie2
 
+    output:
+    file '*.sam' into bowtie2_sam
+
+    script:
+    /* Note! Only read 1 is used for alignment. */
+
+    single = reads instanceof Path
+
+    if(single) {
+      """
+      f='$reads';f=(\$f);f=\${f[0]};f=\${f%.gz};f=\${f%.fastq};f=\${f%.fq};f=\${f%.R1_val_1*};f=\${f%.R2_val_2*};f=\${f%_val_1};f=\${f%_val_2};f=\${f%_trimmed};f=\${f%_1}
+      prefix=\$f
+
+      bowtie2 \\
+        -x $index \\
+        -U $reads \\
+        -k 10 \\
+        --very-sensitive \\
+        -p 1 \\
+        -t \\
+        -S \${prefix}.sam
+      """
+    } else {
+      """
+      f='$reads';f=(\$f);input=\${f[0]};f=\${input%.gz};f=\${f%.fastq};f=\${f%.fq};f=\${f%.R1_val_1*};f=\${f%.R2_val_2*};f=\${f%_val_1};f=\${f%_val_2};f=\${f%_trimmed};f=\${f%_1}
+      prefix=\$f
+
+      bowtie2 \\
+        -x $index \\
+        -U \$input \\
+        -k 10 \\
+        --very-sensitive \\
+        -p 1 \\
+        -t \\
+        -S \${prefix}.sam
+      """
+    }
+}
 
 
 /*
- * STEP 10 - MultiQC
+ * STEP 5.2 - Bowtie 2 post-alignment processing
+ */
+
+process bowtie2_postalignment {
+
+    tag "$name"
+
+    module 'bioinfo-tools'
+    module 'samtools'
+    module 'BEDTools'
+
+    cpus 2
+    memory { 16.GB * task.attempt }
+    time { 120.h * task.attempt }
+    errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+    maxRetries 3
+    maxErrors '-1'
+
+    publishDir "${params.outdir}/bowtie2", mode: 'copy'
+
+    input:
+    file bowtie2_sam
+    set val(name) from name_for_samtools
+
+    output:
+    file '*.sorted.bam' into bowtie2_bam
+    file '*.sorted.bam.bai' into bowtie2_bai
+    file '*.sorted.bed' into bowtie2_bed
+    stdout into bowtie2_logs
+
+    script:
+    """
+    f='$bowtie2_sam';f=(\$f);f=\${f[0]};f=\${f%.sam}
+    prefix=\$f
+
+    samtools view -bT $index \${prefix}.sam > \${prefix}.bam
+    samtools sort \${prefix}.bam \${prefix}.sorted
+    samtools index \${prefix}.sorted.bam
+    rm \${prefix}.sam
+    rm \${prefix}.bam
+    bedtools bamtobed -i \${prefix}.sorted.bam | sort -k 1,1 -k 2,2n -k 3,3n -k 6,6 > \${prefix}.sorted.bed
```

Comment:
What is this used for?

Otherwise, process can be merged with the other two I think.